apps/mobile/src/components/Camera/QrCamera.tsx

```diff
@@ -48,28 +51,67 @@ function CameraFooter(props: { footer: React.ReactNode }) {
   )
 }
 
-function CameraLens({ denied, onPress }: { denied: boolean; onPress: () => Promise<void> }) {
+function CameraLens({
+  denied,
+  onPressSettings,
+  requestPermission,
+  hasPermission,
+  onActivateCamera,
+  isCameraActive,
+}: {
+  denied: boolean
+  onPressSettings: () => Promise<void>
+  requestPermission: () => void
+  hasPermission: boolean
+  onActivateCamera: () => void
+  isCameraActive: boolean
+}) {
+  const handleGrantOrActivatePress = () => {
+    if (!hasPermission) {
+      requestPermission()
+    } else if (hasPermission && !isCameraActive) {
+      onActivateCamera()
+    }
+  }
+
+  const buttonText = denied ? 'Open Settings' : 'Enable camera'
+  const buttonAction = denied ? onPressSettings : handleGrantOrActivatePress
+
   return (
-    <View style={[styles.transparentBox, denied && { backgroundColor: 'rgba(0, 0, 0, 0.8)' }]}>
+    <Pressable
+      style={[styles.transparentBox, denied && { backgroundColor: 'rgba(0, 0, 0, 0.8)' }]}
+      onPress={hasPermission && !isCameraActive ? handleGrantOrActivatePress : undefined}
+      disabled={denied || !hasPermission}
+    >
       {/* Green corners */}
       <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.topLeft]} />
       <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.topRight]} />
       <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.bottomLeft]} />
       <View borderColor={denied ? '$error' : '$success'} style={[styles.corner, styles.bottomRight]} />
 
-      {denied && (
+      {/* Show button/icon only if permission denied, not granted, or granted but inactive */}
+      {(denied || !hasPermission || (hasPermission && !isCameraActive)) && (
         <View style={styles.deniedCameraContainer}>
-          <SafeFontIcon name={'camera'} size={40} color={'$error'} />
-          <SafeButton rounded secondary onPress={onPress} marginTop={20}>
-            Enable camera
+          <SafeFontIcon name={'camera'} size={40} color={denied ? '$error' : '$white50'} />
```

Comment:
You should get some color from the theme and not assign $white50 like this (you can use the helper getTokenValue for example). In dark mode this results to:
![grafik](https://github.com/user-attachments/assets/c115b34f-9f16-460b-af12-3c227a06ed7c)